app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,54 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.all
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show; end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  # POST /recipes or /recipes.json
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = current_user
+
+    respond_to do |format|
+      if @recipe.save
+        format.html { redirect_to recipe_url(@recipe), notice: 'Recipe was successfully created.' }
+        format.json { render :show, status: :created, location: @recipe }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
```

Comment:
[Optional] I think it would be nice if you chose a simple method instead, usually, the user should go to the all recipe page without any initial selection. You can use this approach

```
def create
    @recipe = Recipe.new(recipe_params)
    @recipe.user = current_user

    if @recipe.save
      flash[:notice] = 'Recipe is created!'
      redirect_to recipes_path
    else
      flash[:alert] = 'Recipe is not created!'
      render :new
    end
  end
```